tests/testthat/test-type-date-time.R

```diff
@@ -47,6 +47,13 @@ test_that("vec_proxy() returns a double for Dates with int representation", {
   expect_true(is.double(vec_proxy(x)))
 })
 
+test_that("POSIXlt roundtrips through proxy and restore", {
+  x <- as.POSIXlt("2020-01-03")
+  out <- vec_restore(vec_proxy(x), x)
+  expect_identical(out, x)
```

Comment:
This is why I prefer `expect_identical()` in low level packages by the way. `expect_equal()` would not have checked for row names.